tests/std/syntax/e2eCases/roactToReact/init.luau

```diff
@@ -0,0 +1,271 @@
+
+	-- convert calls
+	local callRepl = query
+		.findallfromroot(ast, queryUtils.isExprCall)
+		:filter(function(node: luau.AstExprCall)
+			local func = node.func
+			if func.tag ~= "indexname" then
+				return false
+			end
+			if not queryUtils.isExprLocal(func.expression) then
+				return false
+			end
+
+			local needsMigration = func.expression.token.text == "Roact" and callMigrations[func.index.text] ~= nil
+			-- specifically catch Roact.Portal
+			local firstCallArg = node.arguments[1].node
+			local isRoactPortal = firstCallArg.tag == "indexname" and firstCallArg.index.text == "Portal"
+			needsMigration = needsMigration or isRoactPortal
+			return needsMigration
+		end)
+		:replace(function(node)
+			local funcName = node.func.index.text
+			local firstCallArg = node.arguments[1].node
+			local isRoactPortal = firstCallArg.tag == "indexname" and firstCallArg.index.text == "Portal"
+
+			if ReactRobloxFunctionTransforms[funcName] or isRoactPortal then
+				needsReactRobloxRequire = true
+			else
+				needsReactRequire = true
+			end
+
+			return if isRoactPortal then callMigrations["Portal"](node) else callMigrations[funcName](node)
+		end)
+
+	-- convert indexname
+	local libIndexNameRepl = utils.transformLibraryAccess(ast, "Roact", function(_, index)
+		return `React.{index.text}`
+	end)
+
+	-- convert index-exprs
+	local libIndexExprRepl = query
+		.findallfromroot(ast, queryUtils.isExprTableItem)
+		:filter(function(entry)
+			if entry.kind ~= "general" or entry.key and entry.key.tag ~= "indexname" then
+				return false
+			end
+
+			local key = entry.key
+			return key.index.text == "Children" or key.index.text == "Ref"
+		end)
+		:replace(function(entry: any) -- had to any caste this to shut up type solver
+			local key = entry.key :: luau.AstExprIndexName
+			local keyRepl
+			if key.index.text == "Children" then
+				keyRepl = ReactPropsTransforms["Roact.Children"](key)
+			elseif key.index.text == "Ref" then
+				keyRepl = ReactPropsTransforms["Roact.Ref"](key)
+			end
+			local entryValue = printer.printnode(entry.value)
+			if keyRepl then
+				needsReactRequire = true
+				-- couldn't replace full entry, so went with this
+				return `{keyRepl} = {entryValue},` :: any
+			end
+			return entry
+		end)
+
+	-- convert/add requires based on all the above (if we need ReactRoblox, if we need React)
+	local addedRequiresRepl = {}
+	if needsReactRequire or needsReactRobloxRequire then
+		local roactRequire = utils.getRequireStatements(ast, "Roact").nodes[1]
+		local requireCall = roactRequire.values[1].node :: luau.AstExprCall
+		local requiredModule = requireCall.arguments[1].node :: luau.AstExprIndexName
+		if needsReactRequire and needsReactRobloxRequire then
+			-- get PackagesRoot
+			local packagesRoot = printer.printnode(requiredModule.expression :: any)
+			addedRequiresRepl[roactRequire] =
+				`local React = require({packagesRoot}.React)\nlocal ReactRoblox = require({packagesRoot}.ReactRoblox)\n`
```

Comment:
This is another spot where it really hurts readability to have the newline in the middle of the format string.